pennylane/measurements/counts.py

```diff
@@ -56,6 +56,12 @@ def counts(op=None, wires=None, all_outcomes=False) -> "CountsMP":
     :math:`p(\lambda_i) = |\langle \xi_i | \psi \rangle|^2`, where :math:`| \xi_i \rangle`
     is the corresponding basis state from the observable's eigenbasis.
 
+    .. note::
+
+        QNodes that return counts cannot, in general, be differentiated, since the derivative
+        with respect to a stochastic process is ill-defined. Please, refer to
+        :func:`~.pennylane.sample` if differentiability is required.
+
```

Comment:
We definitely _could_ support counts if we really wanted to, see [here](https://github.com/PennyLaneAI/pennylane/issues/2932). There's nothing fundamental that's stopping us, other than some (probably) ugly technical details.

Maybe something like this?
```suggestion
    .. note::

        Differentiation of QNodes that return ``counts`` is currently not supported. Please refer to
        :func:`~.pennylane.sample` if differentiability is required.

```

Reply:
Alright. I like it!